sources/packages/backend/libs/integrations/src/services/sfas/sfas-restriction.service.ts

```diff
@@ -45,10 +51,28 @@ export class SFASRestrictionService
         select: { id: true },
         where: { restrictionCode: "LGCY" },
       });
-      await this.repo.manager.query(this.bulkInsertStudentRestrictionsSQL, [
-        legacyRestriction.id,
-        creator.id,
-      ]);
+      await this.dataSource.transaction(async (transactionalEntityManager) => {
+        const bulkInsertLegacyRestrictionsPromise =
+          transactionalEntityManager.query(
+            this.bulkInsertLegacyRestrictionsSQL,
+            [legacyRestriction.id, creator.id],
+          );
+        const bulkInsertSFASMappedRestrictionsPromise =
+          transactionalEntityManager.query(
+            this.bulkInsertSFASMappedRestrictionsSQL,
+            [creator.id],
+          );
+        await Promise.all([
+          bulkInsertLegacyRestrictionsPromise,
+          bulkInsertSFASMappedRestrictionsPromise,
+        ]);
```

Comment:
@andrewsignori-aot Correct me if I'm wrong. I thought we need to execute `bulkInsertLegacyRestrictionsSQL` first and execute `bulkInsertSFASMappedRestrictionsSQL` after. If we put them into parallel with `Promise.all()`, there is a chance that `bulkInsertSFASMappedRestrictionsSQL` could be completed before `bulkInsertLegacyRestrictionsSQL`.

Reply:
@lewischen-aot they are independent sql queries because they don't affect each other and have their own specific inserts. `bulkInsertLegacyRestrictionsSQL` is responsible for inserting Legacy restriction entry if there is none active for that particular student and the student gets one or more restrictions from SFAS which are not present is SIMS. The other query is responsible for inserting all the other restrictions which have an available mapping into SIMS and are either not present in SIMS `student_restrictions` table or are inactive.

Reply:
If they are independent sql, then we can use `Promise.all()`. Thanks for the clarification.